### PR TITLE
Building out documentation

### PR DESCRIPTION
**Pull request by ejh243**
# Description

Initial website development, building out documentation skeleton to facilitate future documentation and repository contributions.

## Issue ticket number

This pull request is to address issues: #17 #21 

## Type of pull request

- [ ] Bug fix
- [ ] New feature/enhancement
- [x] Code refactor
- [x] Documentation update

## Checklist

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have tested my code to check that it is functional
- [X] I have used linters to check for common sources of errors
- [X] I have implemented fail safes in my code to account for edge cases
- [X] I have made the corresponding changes to the documentation

### CODE REVIEW

File: documentation/docs/Contributing/Contributing_to_repository/development-pipeline.md [hunk 1]
```diff
@@ -3,11 +3,11 @@ sidebar_position: 1
 ---
 # The Development Pipeline
```
Comment (ejh243) [inline documentation/docs/Contributing/Contributing_to_repository/development-pipeline.md:4]: @sof202 The word pipeline in this context could get confused with data processing pipeline. I wonder if a title for this page like - "Contributions - A Quick Start Guide" might be more appropriate?

File: documentation/docs/Contributing/Contributing_to_repository/development-pipeline.md [hunk 2]
```diff
@@ -18,7 +18,7 @@ If you are ever unsure about one of the steps described in this document, please
 
 ## Step One: Get the latest version of BrainFANS
```
Comment (ejh243) [inline documentation/docs/Contributing/Contributing_to_repository/development-pipeline.md:19]: Steps 1 & 2 could happen either way around. I have a preference for Issue first as it make this page a more universal guide for how to make a change, whether you came up with the improvement yourself or you picked it up from the issues?

File: documentation/docs/Contributing/Contributing_to_repository/development-pipeline.md [hunk 3]
```diff
@@ -52,7 +52,7 @@ Before you make any changes to the repository, we recommend that you create an i
 ### Select the relevant template for your report
```
Comment (ejh243) [inline documentation/docs/Contributing/Contributing_to_repository/development-pipeline.md:52]: We should encourage checking of existing issues before default creating a new issue. Also creates a pathway whereby people know how to contribute even if they don't have a thing they want to change. Could we generalise this page to include, people who want to make a change but find that change on the existing issue list?

File: documentation/docs/Contributing/Contributing_to_repository/development-pipeline.md [hunk 4]
```diff
@@ -72,10 +72,10 @@ Add relevant personnel to assignees section, assignees will be notified of the i
 
 ## Step Three: Create a new branch
```
Comment (ejh243) [inline documentation/docs/Contributing/Contributing_to_repository/development-pipeline.md:73]: Here we should also encourage checking a relevant branch exists before default creating a new one.

**Review by ejh243 — COMMENTED** [review]
@sof202 Really nice and very well explained guide! Some thoughts on clarifications that might be beneficial.

File: documentation/docs/Contributing/Contributing_to_repository/development-pipeline.md (hunk 1 above)
Comment (sof202) [reply documentation/docs/Contributing/Contributing_to_repository/development-pipeline.md:4]: I agree. This is a better name for it.

File: documentation/docs/Contributing/Contributing_to_repository/development-pipeline.md (hunk 2 above)
Comment (sof202) [reply documentation/docs/Contributing/Contributing_to_repository/development-pipeline.md:19]: That's a good point.

File: documentation/docs/Contributing/Contributing_to_repository/development-pipeline.md (hunk 3 above)
Comment (sof202) [reply documentation/docs/Contributing/Contributing_to_repository/development-pipeline.md:52]: I also thought this, it should be more clear once I separate steps onto their own pages (the description page can then point different people in the correct direction). This was harder to do within a single markdown document (and make it easy to follow)

File: documentation/docs/Contributing/Contributing_to_repository/development-pipeline.md (hunk 4 above)
Comment (sof202) [reply documentation/docs/Contributing/Contributing_to_repository/development-pipeline.md:73]: How would you suggest we explain this? If a relevant branch exists, we should really be pointing towards the user creating a new branch that stems from the relevant branch. I will try to explain this succinctly in a later revision

Comment (ejh243) [reply documentation/docs/Contributing/Contributing_to_repository/development-pipeline.md:73]: Yeah I agree runs the risk of information overload in a "quick guide". Maybe we can ignore here on assumption that when using the quick guide you are likely doing a small independent fix and perhaps creating and then deleting the branch ok?

Comment (sof202) [reply documentation/docs/Contributing/Contributing_to_repository/development-pipeline.md:73]: I could link to an unlisted page that goes into more depth. That way the information will be in a relevant area, but isn't causing information overload.

**Review by sof202 — APPROVED** [review]
Code review completed on feature branch, initial deployment of website is ready to go. There are a few small changes that can be made, but for the most part the pages fulfil their purpose. Website is able to run locally so merge to master is projected to deploy properly.

**Pull request merged**